### PR TITLE
CMake: improve cudart handling

### PR DESCRIPTION
**Pull request by havogt**

### CODE REVIEW

**Comment by havogt** [comment]
launch jenkins

File: tests/src/CMakeLists.txt [hunk 1]
```diff
@@ -13,8 +13,5 @@ endif()
 if (TARGET gcl_gpu)
     add_library(mpi_gtest_main_gpu STATIC EXCLUDE_FROM_ALL mpi_test_driver.cpp)
     target_compile_definitions(mpi_gtest_main_gpu PRIVATE GT_HAS_CUDA)
-    target_link_libraries(mpi_gtest_main_gpu PUBLIC gridtools gtest MPI::MPI_CXX ${GT_CUDA_LIBRARIES})
-    if (GT_CUDA_INCLUDE_DIRS)
-        target_include_directories(mpi_gtest_main_gpu PRIVATE ${GT_CUDA_INCLUDE_DIRS})
-    endif()
+    target_link_libraries(mpi_gtest_main_gpu PUBLIC gridtools gtest MPI::MPI_CXX gcl_gpu)
```
Comment (anstaf) [inline tests/src/CMakeLists.txt:16]: should we also link `mpi_gtest_main_cpu` with `gcl_cpu` ?

**Comment by havogt** [comment]
launch jenkins

**Comment by havogt** [comment]
launch jenkins

File: tests/src/CMakeLists.txt (hunk 1 above)
Comment (havogt) [reply tests/src/CMakeLists.txt]: Actually linking to gcl_gpu was a mistake as it then tries to compile the file as a cuda file...

**Pull request merged**